### PR TITLE
Update quilt packages to support webpack 5

### PR DESCRIPTION
**Pull request by vsumner**
## Description
This PR updates packages with `webpack` support to version 5.

Fixes (issue #)

## Type of change

- [ ] `@shopify/jest-dom-mocks` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] `graphql-mini-transforms` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] `@shopify/react-server` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] `@shopify/sewing-kit-koa` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] `@shopify/web-worker` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/graphql-mini-transforms/package.json [hunk 1]
```diff
@@ -38,17 +38,17 @@
...
   },
   "devDependencies": {
     "@types/common-tags": "^1.8.0",
-    "@types/loader-utils": "^1.1.3",
-    "common-tags": "^1.8.0"
+    "common-tags": "^1.8.0",
+    "webpack": "^5.40.0"
+  },
+  "peerDependencies": {
+    "webpack": ">=5.38.0"
```
Comment (BPScott) [inline packages/graphql-mini-transforms/package.json:51]: Is webpack here used just for types? If that's the case can we mark it as an optional peer dependency? https://github.com/Shopify/loom/blob/06c44051385af276a1982a583b34aeac6160c75f/packages/loom-plugin-babel/package.json#L45-L48

File: packages/graphql-mini-transforms/package.json [hunk 2]
```diff
@@ -38,17 +38,17 @@
...
-    "graphql-typed": "^1.1.1",
-    "loader-utils": "^2.0.0"
+    "graphql-typed": "^1.1.1"
   },
   "devDependencies": {
     "@types/common-tags": "^1.8.0",
-    "@types/loader-utils": "^1.1.3",
-    "common-tags": "^1.8.0"
+    "common-tags": "^1.8.0",
+    "webpack": "^5.40.0"
```
Comment (dahukish) [inline packages/graphql-mini-transforms/package.json:48]: ❤️

**Review by BPScott — APPROVED** [review]
I'm assuming this looks good as I'm very light on webpack context.

One question: Perviously we had had webpack dependencies for `^4.BLAH` (as single major version) and we've widened them up to `>=5.BLAH` so now when webpack 6 comes along these packages will claim to work. Do we feel good about that being the case in webpack 6? (In the past we've done this for lint tooling where we doubt be public API will change much but it seems like when webpack does a major version everything changes.)

**Comment by BPScott** [comment]
Bonsu thought: Whats the migration path like for these packages? Is there anything that consumers need to change as part of bumping to the new major versions

File: packages/sewing-kit-koa/src/assets.ts [hunk 3]
```diff
@@ -44,20 +44,10 @@ export default class Assets {
   }
 
   async scripts(options: AssetOptions = {}) {
-    const js = getAssetsFromManifest(
+    return getAssetsFromManifest(
       {...options, kind: AssetKind.Scripts},
       await this.getResolvedManifest(options.locale),
     );
-
-    const scripts =
```
Comment (BPScott) [inline packages/sewing-kit-koa/src/assets.ts:52]: This seems like a breaking change that should have some attention drawn to it

File: packages/react-server/package.json [hunk 4]
```diff
@@ -52,12 +52,13 @@
...
     "@types/webpack-virtual-modules": "^0.1.0",
     "get-port": "^5.0.0",
+    "memfs": "^3.2.2",
     "node-loader": "^1.0.0",
     "react": "^17.0.2",
     "react-dom": "17.0.2",
     "saddle-up": "^0.5.4",
     "setimmediate": "^1.0.5",
-    "webpack": "^4.25.1"
+    "webpack": ">=5.40.0"
```
Comment (dahukish) [inline packages/react-server/package.json:61]: Should this be 5.38?

File: packages/graphql-mini-transforms/src/webpack-loader.ts [hunk 5]
```diff
@@ -61,7 +59,7 @@ async function loadDocument(
         loader.resolve(resolveContext, imported, (error, result) => {
           if (error) {
             reject(error);
-          } else {
+          } else if (result) {
```
Comment (dahukish) [inline packages/graphql-mini-transforms/src/webpack-loader.ts:62]: Do we still want a default?

File: packages/react-server/src/webpack-plugin/tests/webpack-plugin.test.ts [hunk 6]
```diff
@@ -70,13 +74,16 @@ describe('webpack-plugin', () => {
           await workspace.write('webpack.config.js', BASIC_WEBPACK_CONFIG);
           await workspace.write('client/index.js', BASIC_ENTRY);
 
-          const [serverResults, clientResults] = await runBuild(name);
-          const clientModule = getModule(clientResults, 'client');
-          const serverModule = getModule(serverResults, 'server');
+          const [client, clientIndex, server] = await runBuild(name, [
+            DEFAULT_CLIENT_FILE_PATH,
+            'client/index.js',
```
Comment (dahukish) [inline packages/react-server/src/webpack-plugin/tests/webpack-plugin.test.ts:79]: This seems strange that you pass two client entries and return all three.

File: packages/react-server/src/webpack-plugin/tests/webpack-plugin.test.ts [hunk 7]
```diff
@@ -332,42 +348,38 @@ function runBuild(configPath: string): Promise<any[]> {
           context: pathFromRoot,
         };
 
-    // We use MemoryOutputFileSystem to prevent webpack from outputting to our actual FS
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    const MemoryOutputFileSystem = require('webpack/lib/MemoryOutputFileSystem');
-
     const compiler: Compiler = webpack(contextConfig);
-    compiler.outputFileSystem = new MemoryOutputFileSystem({});
+    // We use memfs.fs to prevent webpack from outputting to our actual FS
```
Comment (dahukish) [inline packages/react-server/src/webpack-plugin/tests/webpack-plugin.test.ts:352]: ❤️

File: yarn.lock [hunk 8]
```diff
@@ -3881,20 +3872,15 @@ acorn-walk@^7.1.1:
...
 acorn@^7.1.1, acorn@^7.2.0:
   version "7.3.1"
   resolved "https://registry.yarnpkg.com/acorn/-/acorn-7.3.1.tgz#85010754db53c3fbaf3b9ea3e083aa5c5d147ffd"
   integrity sha512-tLc0wSnatxAQHVHUapaHdz72pi9KUyHjq5KyHjGg9Y8Ifdc79pTh2XvI6I1/chZbnM7QtNKzh66ooDogPZSleA==
 
-acorn@^8.2.4:
-  version "8.4.0"
-  resolved "https://registry.yarnpkg.com/acorn/-/acorn-8.4.0.tgz#af53266e698d7cffa416714b503066a82221be60"
-  integrity sha512-ULr0LDaEqQrMFGyQ3bhJkLsbtrQ8QibAseGZeaSUiT/6zb9IvIkomWHJIvgvwad+hinRAgsI51JcWk2yvwyL+w==
+acorn@^8.2.4, acorn@^8.4.1:
```
Comment (dahukish) [inline yarn.lock:3880]: We have both acorn 7 and 8?

**Review by dahukish — APPROVED** [review]
Couple of small questions but nothing major. 🏅

File: packages/graphql-mini-transforms/package.json (hunk 1 above)
Comment (vsumner) [reply packages/graphql-mini-transforms/package.json]: Good catch. Updated.

File: packages/react-server/package.json (hunk 4 above)
Comment (vsumner) [reply packages/react-server/package.json]: Yep! Fixed

File: packages/web-worker/src/tests/utilities/webpack.ts [hunk 9]
```diff
@@ -65,7 +65,7 @@ export function runWebpack(
       (error, stats) => {
         if (error) {
           reject(error);
-        } else if (stats.hasErrors()) {
+        } else if (stats && stats.hasErrors()) {
```
[the remark this answers is not in the document]
Comment (vsumner) [reply packages/web-worker/src/tests/utilities/webpack.ts:68]: they can be undefined.

File: yarn.lock [hunk 10]
```diff
@@ -11952,7 +11258,16 @@ read@1, read@~1.0.1:
...
+readable-stream@3, readable-stream@^3.0.0, readable-stream@^3.0.2, readable-stream@^3.1.1, readable-stream@^3.4.0:
+  version "3.6.0"
+  resolved "https://registry.yarnpkg.com/readable-stream/-/readable-stream-3.6.0.tgz#337bbda3adc0706bd3e024426a286d4b4b2c9198"
+  integrity sha512-BViHy7LKeTz4oNnkcLJ+lVSL6vpiFeX6/d3oSH8zCW7UxP2onchk+vTGB143xuFjHS3deTgkKoXXymXqymiIdA==
+  dependencies:
+    inherits "^2.0.3"
+    string_decoder "^1.1.1"
+    util-deprecate "^1.0.1"
+
+readable-stream@^2.0.5, readable-stream@^2.0.6, readable-stream@~2.3.6:
```
[the remark this answers is not in the document]
Comment (vsumner) [reply yarn.lock:11270]: I tried, I haven't been able too

**Pull request merged**